blueprints/geometry/line.py

```diff
@@ -1 +1,229 @@
+        Parameters
+        ----------
+        distance : float
+            Distance to extend or shorten the line. Positive number extends the line, negative number shortens the line.
+        direction: Literal["start", "end"]
+            Given direction where the line needs to be extended. Default towards the end of the line.
+
+        Returns
+        -------
+        Point
+            Extended/shortened point of the line.
+        """
+        if distance < 0 and abs(distance) >= self.length:
+            msg = "When shortening the line, the absolute value of the extra length must be less than the total length of the line."
+            raise ValueError(msg)
+
+        match direction.lower():
+            case "end":
+                new_point = self._end + distance * self._unit_vector
+                self.end_point = Point(new_point)
+            case "start":
+                new_point = self._start - distance * self._unit_vector
+                self.start_point = Point(new_point)
+            case _:
+                msg = "Invalid input for 'direction', use 'start' or 'end'."
+                raise ValueError(msg)
+        return Point(new_point)
+
+    def get_evenly_spaced_points(self, n: int = 2) -> list[Point]:
+        """Return a list of evenly spaced internal points of the line from start to end point with an n number of desired points.
+
+        Parameters
+        ----------
+        n : int
+            Total number of internal points desired. A minimum of 2 points are required.
+        """
+        if not isinstance(n, int):
+            msg = "n must be an integer"
+            raise TypeError(msg)
+        if n < 2:
+            msg = "n must be equal or greater than 2"
+            raise ValueError(msg)
+
+        # Create a list of evenly spaced points
+        evenly_spaced_points = np.linspace(start=0, stop=self.length, num=n, endpoint=True)
+
+        return [Point(self._start + distance * self._unit_vector) for distance in evenly_spaced_points]
+
+    def divide_into_n_lines(self, n: int = 2) -> list["Line"]:
+        """Return a list of evenly divided lines.
+
+        Parameters
+        ----------
+        n : int
+            Total number of lines desired. A minimum of 2 lines are required.
+        """
+        if not isinstance(n, int):
+            msg = "n must be an integer"
+            raise TypeError(msg)
+        if n < 2:
+            msg = "n must be equal or greater than 2"
+            raise ValueError(msg)
+
+        evenly_spaced_points = self.get_evenly_spaced_points(n + 1)
+        return [Line(start_point=point_1, end_point=point_2) for point_1, point_2 in zip(evenly_spaced_points[:-1], evenly_spaced_points[1:])]
+
+    def __eq__(self, other: object) -> bool:
+        """Return True if the lines are equal."""
+        if not isinstance(other, Line):
+            raise NotImplementedError
+        return np.array_equal(self._start, other._start) and np.array_equal(self._end, other._end)
```

Comment:
shouldn't we compare using np.allclose() ? comparing floats can be nasty

Reply:
👍